app/controllers/products_controller.rb

```diff
@@ -2,28 +2,57 @@ class ProductsController < ApplicationController
   before_action :set_product, only:[:show, :edit, :update, :destroy]
   
   def index
+    @products = Product.all
   end
-  
+
   def show
   end
-  
+
   def dashboard
+    @products = Product.all
   end
-  
+
   def new
+    @product = Product.new
+    @products = Product.all
+  end
+
+  def nil_params
+    product_params[:image].blank? || product_params[:description].blank? || product_params[:price].blank? || product_params[:title].blank?
```

Comment:
I would rename this method into something like `validate_params` or `check_params` and I would add a before action in the beginning of the controller:

`before_action :validate_params, only:[:create]`

Also, if you have validations on the model (which I recommend), then you don't need to check for empty parameters since the `create` method will fail if one of the parameter is empty.